### PR TITLE
[Material] Theme-able TextStyles for AlertDialog

### PR DESCRIPTION
**Pull request by johnsonmh**
This PR adds two `TextStyle` parameters to `AlertDialog`s, this allows users to customize the title and content body text of an `AlertDialog`.

Both of these fields were added to the `DialogTheme`, so they are both theme-able.

Note: This change is backwards compatible.

### CODE REVIEW

**Review by willlarche — COMMENTED** [review]
LGTM

File: packages/flutter/lib/src/material/dialog.dart [hunk 1]
```diff
@@ -232,6 +233,11 @@ class AlertDialog extends StatelessWidget {
   /// [actions].
   final EdgeInsetsGeometry titlePadding;
 
+  /// Style for the text in the title of this [AlertDialog].
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/dialog.dart]: title => [title]

File: packages/flutter/lib/src/material/dialog.dart [hunk 2]
```diff
@@ -248,6 +254,11 @@ class AlertDialog extends StatelessWidget {
   /// to separate the content from the other edges of the dialog.
   final EdgeInsetsGeometry contentPadding;
 
+  /// Style for the text in the content section of this [AlertDialog].
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/dialog.dart]: content section => [content]

File: packages/flutter/lib/src/material/dialog.dart [hunk 3]
```diff
@@ -248,6 +254,11 @@ class AlertDialog extends StatelessWidget {
...
+  ///
+  /// If null, defaults to [ThemeData.textTheme.subhead].
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/dialog.dart]: Need to include DialogTheme

File: packages/flutter/lib/src/material/dialog.dart [hunk 4]
```diff
@@ -232,6 +233,11 @@ class AlertDialog extends StatelessWidget {
   /// [actions].
   final EdgeInsetsGeometry titlePadding;
 
+  /// Style for the text in the title of this [AlertDialog].
+  ///
+  /// If null, defaults to [ThemeData.textTheme.title].
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/dialog.dart]: Need to include DialogTheme

File: packages/flutter/lib/src/material/dialog_theme.dart [hunk 5]
```diff
@@ -18,16 +18,29 @@ import 'theme.dart';
 /// When Shape is `null`, the dialog defaults to a [RoundedRectangleBorder] with
 /// a border radius of 2.0 on all corners.
 ///
+/// [titleTextStyle] and [contentTextStyle] are only used in [AlertDialogs].
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/dialog_theme.dart]: are only => are

File: packages/flutter/lib/src/material/dialog_theme.dart [hunk 6]
```diff
@@ -38,13 +51,35 @@ class DialogTheme extends Diagnosticable {
   /// Default value for [Dialog.shape].
   final ShapeBorder shape;
 
+  /// Default value for [AlertDialog.titleTextStyle].
+  ///
+  /// Only used in [AlertDialog]s.
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/dialog_theme.dart]: Used to configure the [DefaultTextStyle] for the [AlertDialog.title] widget.

File: packages/flutter/lib/src/material/dialog_theme.dart [hunk 7]
```diff
@@ -38,13 +51,35 @@ class DialogTheme extends Diagnosticable {
...
+  ///
+  /// If null, defaults to [ThemeData.textTheme.title].
+  final TextStyle titleTextStyle;
+
+  /// Default value for [AlertDialog.contentTextStyle].
+  ///
+  /// Only used in [AlertDialog]s.
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/dialog_theme.dart]: Used to configure the [DefaultTextStyle] for the [AlertDialog.content] widget.

File: packages/flutter/lib/src/material/dialog_theme.dart [hunk 8]
```diff
@@ -88,5 +127,7 @@ class DialogTheme extends Diagnosticable {
     properties.add(DiagnosticsProperty<Color>('backgroundColor', backgroundColor));
     properties.add(DiagnosticsProperty<ShapeBorder>('shape', shape, defaultValue: null));
     properties.add(DiagnosticsProperty<double>('elevation', elevation));
+    properties.add(DiagnosticsProperty<TextStyle>('titleTextStyle', titleTextStyle, defaultValue: null));
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/dialog_theme.dart]: Please update the debugFillProperties test and rename the test `Dialog Theme implements debugFillProperties`

File: packages/flutter/test/material/dialog_test.dart [hunk 9]
```diff
@@ -39,6 +40,11 @@ Material _getMaterialFromDialog(WidgetTester tester) {
   return tester.widget<Material>(find.descendant(of: find.byType(AlertDialog), matching: find.byType(Material)));
 }
 
+RenderParagraph _getTextRenderObjectFromDialog(WidgetTester tester, String text) {
+  final StatelessElement contentWidget = tester.element(find.descendant(of: find.byType(AlertDialog), matching: find.text(text)));
```
Comment (HansMuller) [inline packages/flutter/test/material/dialog_test.dart]: Just return `tester.element<StatelessElement>(...)`

File: packages/flutter/test/material/dialog_theme_test.dart [hunk 10]
```diff
@@ -128,4 +133,38 @@ void main() {
       skip: !Platform.isLinux,
     );
   });
+
+  testWidgets('Custom Title Text Style', (WidgetTester tester) async {
```
Comment (HansMuller) [inline packages/flutter/test/material/dialog_theme_test.dart]: This test and the one below should test the 3 sources for the text style: Theme, DialogTheme, constructor parameter.

**Review by HansMuller — COMMENTED** [review]
This all looks good, just some small stuff.

File: packages/flutter/lib/src/material/dialog.dart (hunk 1 above)
Comment (johnsonmh) [reply packages/flutter/lib/src/material/dialog.dart]: Done.

File: packages/flutter/lib/src/material/dialog.dart (hunk 2 above)
Comment (johnsonmh) [reply packages/flutter/lib/src/material/dialog.dart]: Done.

File: packages/flutter/lib/src/material/dialog.dart (hunk 3 above)
Comment (johnsonmh) [reply packages/flutter/lib/src/material/dialog.dart]: Done.

File: packages/flutter/lib/src/material/dialog.dart (hunk 4 above)
Comment (johnsonmh) [reply packages/flutter/lib/src/material/dialog.dart]: Done.

File: packages/flutter/lib/src/material/dialog_theme.dart (hunk 5 above)
Comment (johnsonmh) [reply packages/flutter/lib/src/material/dialog_theme.dart]: Done.

File: packages/flutter/lib/src/material/dialog_theme.dart (hunk 6 above)
Comment (johnsonmh) [reply packages/flutter/lib/src/material/dialog_theme.dart]: Done.

File: packages/flutter/lib/src/material/dialog_theme.dart (hunk 7 above)
Comment (johnsonmh) [reply packages/flutter/lib/src/material/dialog_theme.dart]: Done.

File: packages/flutter/lib/src/material/dialog_theme.dart (hunk 8 above)
Comment (johnsonmh) [reply packages/flutter/lib/src/material/dialog_theme.dart]: Done.

File: packages/flutter/test/material/dialog_test.dart (hunk 9 above)
Comment (johnsonmh) [reply packages/flutter/test/material/dialog_test.dart]: Done.

File: packages/flutter/test/material/dialog_theme_test.dart (hunk 10 above)
Comment (johnsonmh) [reply packages/flutter/test/material/dialog_theme_test.dart]: I'm not sure if this is what you were asking for, but I wasn't sure how to incorporate all 3 of those things in each test. I split the `'Custom Title Text Style'` and `'Custom Content Text Style'` tests into 3 different tests each. That way I'm testing 
1. Constructor Param
1. Dialog Theme
1. Theme.textTheme

For both `titleTextStyle` and  `contentTextStyle` (a total of 6 tests).

Comment (HansMuller) [reply packages/flutter/test/material/dialog_theme_test.dart]: I would have written two tests like this:
```
testWidgets('Custom Title Text Style', (WidgetTester tester) async {
  Widget buildFrame({ TextStyle themeStyle, TextStyle dialogThemeStyle, TextStyle dialogStyle }) {
    return _appWithAlertDialog(
      tester,
      AlertDialog(
        title: Text(titleText),
        titleTextStyle: titleTextStyle,
        actions: <Widget>[ ],
      ),
      theme: ThemeData(
        textTheme: const TextTheme(title: themeStyle)
        dialogTheme: const DialogTheme(titleTextStyle: dialogStyle)
      ),
    );
  }

  await tester.pumpWidget(
    buildFrame(
      themeStyle: const TextStyle(color: Colors.pink),
      dialogThemeStyle: const TextStyle(color: Colors.green),
      dialogStyle: const TextStyle(color: Colors.blue),
    ),
  );
  await tester.tap(find.text('X'));
  await tester.pumpAndSettle();
  final RenderParagraph title = _getTextRenderObjectFromDialog(tester, titleText);
  expect(title.text.style, const TextStyle(color: Colors.blue));

  await tester.pumpWidget(
    buildFrame(
      themeStyle: const TextStyle(color: Colors.pink),
      dialogThemeStyle: const TextStyle(color: Colors.green),
    ),
  );
  await tester.tap(find.text('X'));
  await tester.pumpAndSettle();
  final RenderParagraph title = _getTextRenderObjectFromDialog(tester, titleText);
  expect(title.text.style, const TextStyle(color: Colors.green));

  await tester.pumpWidget(
    buildFrame(
      themeStyle: const TextStyle(color: Colors.pink),
    ),
  );
  await tester.tap(find.text('X'));
  await tester.pumpAndSettle();
  final RenderParagraph title = _getTextRenderObjectFromDialog(tester, titleText);
  expect(title.text.style, const TextStyle(color: Colors.pink));

});
```

**Review by HansMuller — APPROVED** [review]
LGTM, although you could make rewrite 6 TextStyle tests as 2.

File: packages/flutter/test/material/dialog_theme_test.dart (hunk 10 above)
Comment (johnsonmh) [reply packages/flutter/test/material/dialog_theme_test.dart]: ohh I see now, in general I personally prefer to opt for smaller tests: go/tott-520

**Pull request merged**